docs/environment-setup.md

```diff
@@ -74,94 +77,190 @@ The `docker exec` command executes a command inside of the Docker container.
+- [`docker stop`](https://docs.docker.com/engine/reference/commandline/stop/) gracefully stops a running container
+- [`docker kill`](https://docs.docker.com/engine/reference/commandline/kill/) aborts a running container
+
+## macOS
+
+Core OFRAK can be run locally on macOS.
+
+It is recommended that you create a virtual environment in which to install the code:
+
+```bash
+python3 -m venv ofrak-venv
+source ofrak-venv/bin/activate
+```
 
+1. Use homebrew to install required libraries and executables:
+
+    ```bash
+    brew install apktool binwalk cmake java libmagic lzop p7zip qemu squashfs rar unar wget
+    ```
+
+    - OFRAK uses `apktool`, `java`, and `wget` to install `uber-apk-signer` for unpacking and packing APK files.
+    - OFRAK uses `binwalk` for analyzing packed binary files.
+    - OFRAK uses `cmake` to install a custom branch of `keystone-engine`.
+    - OFRAK uses the `libmagic` library for `python-magic`, which automatically determines which packers/unpackers to use with binaries.
+    - OFRAK uses the `lzop` command line utility for packing/unpacking LZO archives
+    - OFRAK uses the 7-zip command line utility for packing/unpacking 7z archives
+    - OFRAK uses the `qemu-system-i386` command line utility (and other `qemu` commands) for testing the `BzImage` packer and unpacker
+    - OFRAK uses the `mksquashfs` command line utility for packing/unpacking SquashFS filesystems.
+    - OFRAK uses the `rar` and `unar` command line utilities for packing/unpacking RAR archives
+
+    If not all of the dependencies are installed, core OFRAK will still work, but most of the components will not.
+
+2. The following script can then be used to install `keystone`:
+
+    ```bash
+    #!/usr/bin/env bash
+    set -e
+    pushd /tmp
+    git clone https://github.com/rbs-forks/keystone.git
+    cd keystone
+    git checkout 2021.09.01
+    mkdir build
+    cd build
+    ../make-share.sh
+    cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=ON -G "Unix Makefiles" ..
+    make -j8
+    sudo make install
+    cd ../bindings/python
+    make install3
+    pip3 install .
+    popd
+    ```
+3. The custom branch of `capstone` can be installed using the following script:
+
+    ```bash
+    #!/usr/bin/env bash
+    set -e
+    pushd /tmp
+    git clone https://github.com/rbs-forks/capstone.git
+    cd capstone
+    git checkout 2021.09.01
+    ./make.sh
+    sudo ./make.sh install
+    cd bindings/python
+    make install3
+    popd
+    ```
+4. The `uber-apk-signer` for unpacking and packing APK files can be installed using the following script:
+
+    ```bash
+    brew install wget apktool java
+    echo 'export PATH="/usr/local/opt/openjdk/bin:$PATH"' >> ~/.zshrc
+    wget https://github.com/patrickfav/uber-apk-signer/releases/download/v1.0.0/uber-apk-signer-1.0.0.jar -O /usr/local/bin/uber-apk-signer.jar
+    ```
+5. Install core OFRAK and its dependencies:
+
+    ```bash
+    INSTALL_TARGET=develop
+    make -C ofrak_core $INSTALL_TARGET
```

Comment:
Does this work?

One probably needs to install:
- ofrak_io
- ofrak_type
- ofrak_patch_maker
- ofrak_components
- ofrak